apps/scratch/src/vm/load-crt-project.ts

```diff
@@ -1,11 +1,41 @@
 import VM from "scratch-vm";
 import JSZip from "jszip";
-import { ScratchCrtConfig } from "../types/scratch-vm-custom";
+import { ScratchCrtConfig, ScratchProject } from "../types/scratch-vm-custom";
 import { defaultCrtConfig } from "./default-crt-config";
 
 let nextProject: ArrayBuffer | undefined = undefined;
 let isLoading = false;
 
+/**
+ * Check if all assets listed in project.json exist in the ZIP
+ * @param zip The JSZip instance representing the loaded ZIP file.
+ * @param project The parsed project.json object.
+ * @return A list of missing asset filenames. Empty if all assets are present.
+ */
+const checkAssets = async (
```

Comment:
If possible, name a function according to what it does. Here we can be much more specific like:

```suggestion
const getMissingFilenames = async (
```